flake.nix

```diff
@@ -422,6 +424,40 @@
             "modules/fedimint-wallet"
           ];
         };
+
+        # Replace placeholder git hash in a binary
+        #
+        # To avoid impurity, we use a git hash placeholder when building binaries
+        # and then replace them with the real git hash in the binaries themselves.
+        replace-git-hash = { package, name }:
+          let
+            # the git hash placeholder we use in `build.rs` scripts when
+            # building in Nix (to preserve purity)
+            hash-placeholder = "01234569afbe457afa1d2683a099c7af48a523c1";
+            # the hash we will set if the tree is dirty;
+            dirty-hash = "0000000000000000000000000000000000000000";
+            # git hash to set (passed by Nix if the tree is clean, or `dirty-hash` when dirty)
+            git-hash = if (self ? rev) then self.rev else dirty-hash;
+          in
+          pkgs.stdenv.mkDerivation {
+            inherit system;
+            inherit name;
+
+            dontUnpack = true;
+
+            installPhase = ''
+              cp -a ${package} $out
+              for path in `find $out -type f -executable`; do
+                bbe -e 's/${hash-placeholder}/${git-hash}/' $path > ./tmp || exit 1
+                chmod +w $path
+                cat ./tmp > "$path"
```

Comment:
why do we `cat` here? to see whats going on ?

Reply:
To keep the original permission etc. as they were and only replace the content.

Reply:
oh I didn't even see it was redirected

Reply:
Added comment.